test/test_torch.py

```diff
@@ -4548,38 +4548,63 @@ def test_index_fill(self, device, dtype):
-        for dtype in [torch.int, torch.long]:
-            src = torch.randn(3, 4, 5, device=device)
-            # Index can be duplicated.
-            idx = torch.tensor([2, 1, 0, 1, 2], dtype=dtype, device=device)
-            dest = torch.index_select(src, 0, idx)
-            self.assertEqual(dest.shape, (5, 4, 5))
-            for i in range(idx.size(0)):
-                self.assertEqual(dest[i], src[idx[i]])
-
-            # Check that 'out' is used correctly.
-            out = torch.randn(5 * 4 * 5, device=device)
-            dest = torch.index_select(src, 0, idx, out=out.view(5, 4, 5))
-            self.assertEqual(dest.shape, (5, 4, 5))
-            for i in range(idx.size(0)):
-                self.assertEqual(dest[i], src[idx[i]])
-            out.fill_(0.123)
-            self.assertEqual(out, dest.view(-1))  # Must point to the same storage.
-
-            # Bool tensor
-            src = torch.tensor([False, True, False, False], device=device, dtype=torch.bool)
-            idx = torch.tensor([1], dtype=dtype, device=device)
-            dest = torch.index_select(src, 0, idx)
-            self.assertEqual(torch.tensor([True]), dest)
-
-            # Complex Tensor
-            src = torch.randn(3, 4, 5, dtype=torch.complex64, device=device)
-            idx = torch.tensor([2, 1, 0, 1, 2], dtype=dtype, device=device)
-            dest = torch.index_select(src, 0, idx)
-            self.assertEqual(dest.shape, (5, 4, 5))
-            for i in range(idx.size(0)):
-                self.assertEqual(dest[i], src[idx[i]])
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_index_select(self, device, dtype):
+        num_src, num_dest = 3, 5
+
+        def make_arg(batch_sizes, n, dim, contig):
+            size_arg = batch_sizes[:dim] + (n,) + batch_sizes[dim:]
+            return make_tensor(size_arg, device, dtype, low=None, high=None, discontiguous=not contig)
+
+        def ref_index_select(src, dim, idx):
+            dest_size = list(src.size())
+            dest_size[dim] = len(idx)
+            dest = torch.empty(*dest_size, dtype=src.dtype, device=src.device)
+            for i in range(len(idx)):
+                idx_dest = dim * (slice(None),) + (i,)
+                idx_src = dim * (slice(None),) + (idx[i],)
+                dest[idx_dest] = src[idx_src]
+            return dest
+
+        for src_contig, idx_contig, out_contig in product([True, False], repeat=3):
+            for other_sizes in ((), (4, 5)):
+                for dim in range(len(other_sizes)):
+                    src = make_arg(other_sizes, num_src, dim, src_contig)
+                    idx = make_tensor((num_dest,), device, dtype=torch.int64, low=0, high=num_src, discontiguous=not idx_contig)
+                    dest = torch.index_select(src, dim, idx)
+                    dest2 = ref_index_select(src, dim, idx)
+                    self.assertEqual(dest, dest2)
+
+                    # Out version points to the same tensor
```

Comment:
We recently added testing for out= to the tests generated automatically for each OpInfo; you probably don't need to do any testing of the out= variant here

Reply:
This is for the forward though. Do you still want me to remove it?

Reply:
Yep, we now test the foward out= behavior using sample inputs. Although this did inspire a slight update to that test that I'm working on adding now.